src/controllers/user.controller.js

```diff
@@ -86,4 +87,35 @@ export class UserController {
       });
     }
   }
-}
+    static async getProfile(req, res){
+        try {
+            const user = await UserService.getUserById(req.user.id);
+            return res.status(200).json({
+                message: 'profile gotten successful',
+                data:user
+            });
+        }catch (e){
+            return res.status(500).json({error:'something went wrong', data:e,});
+        }
+    }
+    static async updateProfile(req, res){
+        try{
+            let payload = {
+                ...req.body,
+                billingAddress:JSON.parse(req.body.billingAddress) || {},
+                birthdate:new Date(req.body.birthdate || ''),
+            };
+            if(req.files?.avatar){
+                const {url} = await uploadPhoto(req,res,req.files.avatar);
+                payload['avatar'] = url;
+            }
+            await UserService.updateUser(req.user.id, payload);
+            const updatedUser = await UserService.getUserById(req.user.id);
+            // eslint-disable-next-line no-unused-vars
+            const {dataValues:{password,...others}} = updatedUser;
+            return res.status(200).json({ message: 'updated successful', data: others });
```

Comment:
Since we have endpoint to return user profile, return only message after updating profile